ecmascript/codegen/src/lib.rs

```diff
@@ -901,10 +901,10 @@ impl<'a> Emitter<'a> {
 
         self.emit_accesibility(n.accessibility)?;
 
-        if n.readonly {
-            keyword!("readonly");
-            space!();
-        }
+        // if n.readonly {
+        //     keyword!("readonly");
+        //     space!();
+        // }
```

Comment:
This is probably not right, @kdy1 could you give me a short explanation on how emitter works? `readonly` should be emitted if we emit TypeScript, but should be stripped when emitting JavaScript. I can't find any place where that's decided.

Reply:
@bartlomieju It's determined by the data of node.

If you want special behavior only for typescript, you can change swc_emca_transforms::typescript::strip. In this case, you can modify `fold_private_prop` to remove readonly flag.

Reply:
@bartlomieju If you are ok with it, I'll add some commits to the pr to fix the bug.

Reply:
@kdy1 się, however it seems more bugs were found: 
- It's not just private props: target/debug/deno cache --unstable --no-check -r std/mime/multipart.ts. (The source line is even more haywire here, it's caused by readonly left on public MultipartReader fields.)

- I think there are way more bugs than with readonly.
- target/debug/deno cache --unstable --no-check -r std/path/win32.ts: generates with messed up backslash escape in template literal.
- target/debug/deno cache --unstable --no-check -r std/node/module.ts: generates with messed up hexadecimal escape sequence.

- - target/debug/deno cache --unstable --no-check -r std/path/_globrex.ts: backslash problem described earlier.
- target/debug/deno cache --unstable --no-check -r std/wasi/snapshot_preview1.ts: same.
- target/debug/deno cache --unstable --no-check -r std/ws/test.ts: doesn't strip enum.
That's all the errors I could yield in std, some might be blocked by others.

I'll add tests for all of them in a couple hours, and your commits are more than welcome

Reply:
I fixed private property issue and there are some known - although no issue exists on github - escape issues with codegen.
An enum not being stripped is bit surprising.

Test codes are in the deno repository, is it correct?

Reply:
@bartlomieju  There was a bug in the parser and codegen. I send a [pr](https://github.com/bartlomieju/swc/pull/1)

Reply:
Thanks for explanation @kdy1, I'll add more tests to codegen for the escape sequences.